### DISCUSSION TITLE
Create 3D printable models based on PlanetaryComputers ALOS DEM

### DISCUSSION THREAD

**Discussion by fgebhart** [microsoft/PlanetaryComputer#32] [Show and tell]
Hey there 👋 

I just published a little tool I built recently, called `mapa`. It is based on ALOS DEM catalogue of the Planetary Computer.

`mapa` let's you easily create 3d printable models based on 30m resolution satellite data. The 3d printed models could be used either for decoration or teaching.
Feel free to check it out: https://github.com/fgebhart/mapa

Happy to receive some feedback 🙂

**Comment by TomAugspurger** [comment]
This is fantastic, thanks for sharing :) 

It looks like there's a jupyterlab extension for rendering STL files at https://github.com/pschatzmann/jupyterlab-viewer-3d.

![image](https://user-images.githubusercontent.com/1312546/153080590-5c3e7d9f-e0a8-4b10-ad1a-979bf93a0d96.png)

It took a while to load the SLT file, but it eventually made it.

It's kinda cumbersome to install jupyterlab extensions at runtime (you need nodejs and have to reload jupyterlab) so I'll look into adding this to the base image.

---

We're also happy for feedback with how things went and whether you ran into any pain points.

**Comment by fgebhart** [comment]
Thanks for the feedback! Wasn't aware of the 3d rendering jupyter extension, will also give it a try! :)

With respect to feedback, I would say everything went quite smoothly overall.

I struggled a bit with understanding the need of signing assets until I eventually figured out that I don't need it for my use case at all. It basically was the first time working with geospatial data for me, but I have to say STAC APIs are really great!

Also, it took me some time to figure out an easy way of merging geotiffs. I assumed this to be a super common thing to do. In the end I found `rasterio.merge` which solves this task, even though I find the approach not very straightforward. But that is probably not directly related to planetary computer. 

Will definitely spend some more time interacting with the pc and its resources. I'm especially keen to check out the hub and maybe even scaling my application using dask :)
